myflights/apps/core/scripts/import_data.py

```diff
@@ -0,0 +1,164 @@
+import sys
+import json 
+from django.core.exceptions import ValidationError
+from abc import ABC, abstractproperty
+from pathlib import Path
+
+from ..models import Airline, Airport, Route
+
+
+class BaseImporter(ABC):
+    """Abstract class that provides interface and base functionality
+    to import data from JSON to database.
+    """
+    Model = None
+    batch_size = 100
+
+    def __init__(self):
+        """Initializes `BaseImporter` instance.
+        """
+
+        self.file = Path('data') / self.filename
+        self.new_objects = []
+
+    def _log(self, msg):
+        """Naive log to strerr using built-in print.
+
+        :param msg: message to be logged
+        :type msg: str
+        """
+
+        print(f'[{self.Model}] {msg}', file=sys.stderr)
+
+    @abstractproperty
+    def filename(self):
+        """Abstract property. Provide JSON filename in data directory to be imported.
+        """
+        pass
```

Comment:
To override it in subclasses and provide specific filename.

Reply:
Ok, then at least set it to: 

`raise NotImplementedError("Please Implement this method")`

It will be less error prone this way. Remember, that due to duck typing this abstract method is not needed anyway.

If you really want to do this in more of a java way, you could explore `abc` library, which lets you use `@abc.abstractmethod` (or something similar) decorator enabling you to enforce, that all derived classes will implement what you want. ;)

Reply:
This exactly what I did: `abc` library and `abstractproperty`. :)

Reply:
and I've checked that `abstractproperty` raise exception if someone forgets to override abstract method or property.